src/skydb.ts

```diff
@@ -100,6 +100,21 @@ export async function getJSON(
   // Lookup the registry entry.
   const getEntryOpts = extractOptions(opts, defaultGetEntryOptions);
   const { entry }: { entry: RegistryEntry | null } = await this.registry.getEntry(publicKey, dataKey, getEntryOpts);
+
+  // Calculate the new revision and get the cached revision.
+  const revision = entry?.revision ?? BigInt(-1);
```

Comment:
nit: Defaulting to -1 for revision is a little confusing at first. When I thought about it and figured out it was meant to mean "this has never been set" and to make `prev_rev + 1` useful then it made sense. But it would be nice to move it to a `const` and explain the assumptions.